lib/services/samples-service.ts

```diff
@@ -160,15 +160,32 @@ export class SamplesService implements ISamplesService {
 		}).future<Sample[]>()();
 	}
 
-	private getRepositories(gitHubEndpointUrl: string, filter: (repo: any) => boolean): IFuture<string[]> {
+	private getPagedResult(gitHubEndpointUrl: string, page: number): IFuture<string[]> {
 		return (() => {
 			try {
-				var repos = JSON.parse(this.$httpClient.httpRequest(gitHubEndpointUrl).wait().body);
+				var requestUrl = gitHubEndpointUrl + "&page=" + page.toString();
+				var result = JSON.parse(this.$httpClient.httpRequest(requestUrl).wait().body);
+				return result;
 			} catch (error) {
 				this.$logger.debug(error);
 				this.$errors.fail(SamplesService.SAMPLES_PULL_FAILED_MESSAGE);
 			}
 
+		}).future<string[]>()();
+	}
+
+	private getRepositories(gitHubEndpointUrl: string, filter: (repo: any) => boolean): IFuture<string[]> {
+		return (() => {
+			var repos: string[] = [];
+
+			for (var page = 1; ; ++page) {
+				var pagedResult = this.getPagedResult(gitHubEndpointUrl, page).wait();
+				if(_.isEmpty(pagedResult)) {
+					break;
+				}
+				Array.prototype.push.apply(repos, pagedResult);
```

Comment:
I also think push + _.flatten is easier to read, but maybe it will require casting (as we know _.flatten has the great behavior of losing the types.
So I vote :+1: for the current code.